lib/chef/provider/package/yum.rb

```diff
@@ -51,17 +51,6 @@ def yum_binary
         # Extra attributes
         #
 
-        def arch_for_name(n)
-          if @new_resource.respond_to?("arch")
-            @new_resource.arch
-          elsif @arch
-            idx = package_name_array.index(n)
```

Comment:
We search by name here and find the first occurrence.  We want the correct occurrence but have no way to identify which entry with a given name is the right one.